originations/services/triggers_state_store/store.py

```diff
@@ -2,44 +2,34 @@
 from datetime import datetime
 from originations.enums.policy import PolicyPhase
 from originations.services.firestore.firestore import load_firestore
-from originations.models.past_triggers import PastTriggers
+from originations.models.past_triggers import PastPolicyTrigger
 
 
-async def save_policy_triggers(hash: str, phase: str, triggers: list[str]):
+async def save_policy_triggers(hash: str, phase: str, triggers: list[str]) -> None:
     firestore = await load_firestore()
 
-    reference = (
-        firestore.collection("triggers_store")
-        .document(hash)
-        .collection(phase)
-        .document("latest")
-    )
+    reference = firestore.collection("triggers_state_store").document(hash)
 
-    await reference.set({"event_time": datetime.now(), "triggers": triggers})
+    body = {rule: {"event_time": datetime.now(), "phase": phase} for rule in triggers}
 
+    await reference.set(body, merge=True)
```

Comment:
Now each applicant has a document that contains key:value pairs for each policy rule.

We use Firebase's merge capability here to only update triggers when they happen and leave any existing ones untouched. This means that the document always maintains a record for each rule the most recent time it was triggered (but not historically).

This should cover most use cases!